### DISCUSSION TITLE
GetElementById and GetElementsByTagName

### DISCUSSION THREAD

**Discussion by Mephisto090** [cefsharp/CefSharp.Dom#16] [Q&A] [answered]
```
<div id="test_id_1">
      <strong class="class_test">Strong Text 1</strong>
</div>
<div id="test_id_2">
      <strong class="class_test">Strong Text 2</strong>
</div>
```

I would now like to have the Strong Text 2 output to me. 

According to the principle of the web browser: `GetElementById(TagID).GetElementsByTagName(TagName)`

How do I replace the question marks?

```
Await _devtoolsContext.QuerySelectorAllAsync(???)
Await _devtoolsContext.QuerySelectorAsync(???)
```

**Comment by amaitland** [comment, marked as answer]
Use QuerySelectorAsync for a single element and QuerySelectorAllAsync for multiple. 

https://developer.mozilla.org/en-US/docs/Web/API/Document_object_model/Locating_DOM_elements_using_selectors

**Reply by Mephisto090** [reply]
> Use QuerySelectorAsync for a single element and QuerySelectorAllAsync for multiple.

I'm aware
I mean what the question marks should look like as I work my way down from the top
Example from the web browser:

`GetElementById("test_id_2").GetElementsByTagName("strong")`

Does it have to look like this?
```
Await _devtoolsContext.QuerySelectorAllAsync("div #test_id_2, strong")
Await _devtoolsContext.QuerySelectorAsync("div #test_id_2, strong")
```
Or
```
Await _devtoolsContext.QuerySelectorAllAsync("div #test_id_2, strong.class_test")
Await _devtoolsContext.QuerySelectorAsync("div #test_id_2, strong.class_test")
```

**Reply by amaitland** [reply]
Try this article https://www.javascripttutorial.net/javascript-dom/javascript-queryselector/

**Reply by Mephisto090** [reply]
Just for my understanding, would that be correct in my example above?

```
Await _devtoolsContext.QuerySelectorAllAsync("div#test_id_2 > strong")
Await _devtoolsContext.QuerySelectorAsync("div#test_id_2 > strong")
```

or if I want it to be much clearer like this:
```
Await _devtoolsContext.QuerySelectorAllAsync("div#test_id_2 > strong.class_test")
Await _devtoolsContext.QuerySelectorAsync("div#test_id_2 > strong.class_test")
```

**Reply by amaitland** [reply]
"#test_id_2 strong" should be sufficient based on the example provided.
